.github/workflows/automerge.yml

```diff
@@ -1,23 +1,43 @@
-name: automerge
+name: Scheduled Automerge Master -> Staging -> Develop
 
 on:
   schedule:
-    - cron: '30 0 * * *' # every night
+    - cron: '0 0 * * 1-5' # At 00:00, Monday through Friday
   workflow_dispatch: # on button click
 
 jobs:
-  merge-staging-to-dev:
+  merge-master-to-staging:
+    name: Merge Master to Staging
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - name: Set Git config
         run: |
-          git config --local user.email "${GITHUB_ACTOR}"
-          git config --local user.name "${GITHUB_ACTOR}@users.noreply.github.com"
-      - name: Merge staging to develop
+          git config --local user.email "Sofie-Automation@users.noreply.github.com"
+          git config --local user.name "GitHub Actions # Sofie-Automation"
+      - name: Merge 'master' to 'staging'
+        run: |
+          git fetch --unshallow
+          git checkout staging
+          git pull
+          git merge --no-ff origin/master -m "Automerge 'master' to 'staging'"
+          git push
+
+  merge-staging-to-develop:
+    name: Merge Staging to Develop
+    runs-on: ubuntu-latest
+    needs: merge-master-to-staging
```

Comment:
Will this stop the staging to develop merge if there has only been new changes to staging at not master?

Reply:
The 'merge-master-to-staging' still runs, it just says that 'Everything up-to-date', followed by the 'merge-staging-to-develop'. I just tested it on my own repository i set up for precisely testing these kind of things :)